docs/zkapps/o1js/smart-contracts.md

```diff
@@ -183,36 +148,54 @@ class HelloWorld extends SmartContract {
 }
 ```
 
-The `increment()` method fetches the current on-chain state `x` with `this.x.get()`.
+The `@increment()` method fetches the current on-chain state `x` with `this.x.get()`.
 Later, it sets the new state to `x + 1` using `this.x.set()`. Simple!
 
-There's another line though, which looks weird at first:
+Another line might looks weird at first:
 
 ```ts
 this.x.requireEquals(x);
 ```
 
-To understand it, we have to take a step back, and understand what it means to "use an on-chain value" during off-chain execution.
+Here's what it means to "use an on-chain value" during off-chain execution.
 
-For sure, when we use an on-chain value, we have to _prove_ that this is the on-chain value. Verification has to fail if it's a different value! Otherwise, a malicious user could modify o1js and make it just use any other value than the current on-chain state – breaking our zkApp.
+When you use an on-chain value, you have to _prove_ that this value is the on-chain value. Verification has to fail if it's a different value. Otherwise, a malicious user could modify o1js and make it just use any other value than the current on-chain state – breaking the zkApp.
```

Comment:
our style is to write for the developer and use second person pronoun "you" so I updated all of the singular "we"

Reply:
Sounds good